dune/gdt/local/numerical-fluxes/upwind.hh

```diff
@@ -52,24 +58,40 @@ public:
 
   using BaseType::apply;
 
-  StateRangeType apply(const StateRangeType& u,
-                       const StateRangeType& v,
-                       const PhysicalDomainType& n,
-                       const XT::Common::Parameter& param = {}) const override final
+  StateType apply(const I& intersection,
+                  const LocalIntersectionCoords& x,
+                  const StateType& u,
+                  const StateType& v,
+                  const PhysicalDomainType& n,
+                  const XT::Common::Parameter& param = {}) const override final
   {
-    const auto df = this->flux().jacobian((u + v) / 2., param);
```

Comment:
Same as above. Since this happens everywhere, that we need a `local_flux_inside` and `local_flux_outside`, perhaps these could be members of the interface. In addition to `compute_entity_coords`, the interface could have a protected `bind(intersection)` method, where the mutable inside and outside local fluxes are bound to the respective elements, such that after calling `compute_entity_coords`, the element coordinates match the respective flux.

Reply:
Yes, sounds good, done.